### ISSUE TITLE
Missing files in solution explorer after using Include in Project

### ISSUE THREAD

**Issue by huguesv** [Microsoft/PTVS#2422]
I have a python application project, where I've copied the ptvsd folder while VS was closed.
I open VS, select project node and enable Show all files.
Screenshot shows how it appears, and that ptvsd subfolder has Python files in it.
![image](https://cloud.githubusercontent.com/assets/1696845/25212677/a016e420-2540-11e7-8aab-512b0b972ebf.png)
Right click on ptvsd and Include in Project
![image](https://cloud.githubusercontent.com/assets/1696845/25212694/c289c464-2540-11e7-81cc-508cbc924413.png)
Now if I close the project and reopen (keep show all files enabled), now it shows the files under ptvsd.
![image](https://cloud.githubusercontent.com/assets/1696845/25212728/fa722a1a-2540-11e7-9774-da9d3c43572f.png)

I would expect the files under ptvsd to appear the first time I enabled Show All Files.  Include in Project should have included the folder and all its files in one operation.

That's how I remember it working. Looks like a regression.

**Comment by zooba** [comment]
Is this a reliable repro? Occasionally things can race, which leads to the search failing (I think there's already a bug for this case)

**Comment by huguesv** [comment]
Yes, it's been reliable.  I've copied the project to `\bugs\2422` folder.

**Comment by huguesv** [comment]
I can't repro anymore, so it's either unreliable or some other project system fix that went it took care of it. I'll do a bit more testing before closing.

**Comment by huguesv** [comment]
While I couldn't repro on my dev machine, on an install of the latest 15.5 preview, I was able to repro #2874 but not this one.

**Comment by huguesv** [comment]
I've done extensive testing on my machine where I got the repro.
I copied the release build binaries for Core, and couldn't repro.
I went back and forth between that and the 15.5 preview 2 build, and it consistently only repro'd on the old build.
I also used our powershell script and installed 17261 (15.5 preview 2) and 17297 (today's) and got the same consistent repro with the old build but not the new one.
I also tested the latest build with and without the experimental IntelliSense, just in case but that made no difference.
I'm declaring this one (and #2874) fixed.
